src/client/caching/structures/guilds/Guild.ts

```diff
@@ -402,25 +523,219 @@ export class Guild extends Structure {
+	 */
+	explicit_content_filter?: GuildExplicitContentFilterLevel;
+
+	/**
+	 * The id for afk {@link VoiceChannel channel}.
+	 * @since 0.0.1
+	 */
+	afk_channel_id?: string;
+
+	/**
+	 * The afk timeout in seconds.
+	 * @since 0.0.1
+	 */
+	afk_timeout?: number;
+
+	/**
+	 * The base64 1024x1024 png/jpeg/gif image for the guild icon (can be animated gif when the guild has `ANIMATED_ICON` feature).
+	 * @since 0.0.1
+	 */
+	icon?: string;
+
+	/**
+	 * The {@link User user} id to transfer guild ownership to (must be owner).
+	 * @since 0.0.1
+	 */
+	owner_id?: string;
+
+	/**
+	 * The base64 16:9 png/jpeg image for the guild splash (when the guild has `INVITE_SPLASH` feature).
+	 * @since 0.0.1
+	 */
+	splash?: string;
+
+	/**
+	 * The base64 16:9 png/jpeg image for the guild banner (when the guild has `BANNER` feature).
+	 * @since 0.0.1
+	 */
+	banner?: string;
+
+	/**
+	 * The id of the {@link TextChannel channel} where guild notices such as welcome messages and boost events are posted.
+	 * @since 0.0.1
+	 */
+	system_channel_id?: string;
+
+	/**
+	 * The id of the channel where guilds display rules and/or guidelines (when the guild has `PUBLIC` feature).
+	 * @since 0.0.1
+	 */
+	rules_channel_id?: string;
+
+	/**
+	 * The id of the channel where admins and moderators of guilds receive notices from Discord (when the guild has `PUBLIC` feature).
+	 * @since 0.0.1
+	 */
+	public_updates_channel_id?: string;
+
+	/**
+	 * The preferred locale of a guild used in server discovery and notices from Discord; defaults to "en-US" (when the guild has `PUBLIC` feature).
+	 * @since 0.0.1
+	 */
+	preferred_locale?: string;
 }
 
-interface GuildPruneOptions {
+/**
+ * The options for a dry {@link Guild#prune}.
+ * @since 0.0.1
+ * @see https://discord.com/developers/docs/resources/guild#get-guild-prune-count-query-string-params
+ */
+export interface GuildPruneDryOptions {
+	/**
+	 * The number of days to count prune for (1 or more).
+	 * @since 0.0.1
+	 * @default 7
+	 */
 	days: number;
+
+	/**
+	 * The {@link Role role} IDs to include.
+	 * @since 0.0.1
+	 * @default []
+	 */
+	include_roles: string[];
+}
+
+/**
+ * The options for a non-dry {@link Guild#prune}.
+ * @since 0.0.1
+ * @see https://discord.com/developers/docs/resources/guild#begin-guild-prune-query-string-params
+ */
+export interface GuildPruneOptions extends GuildPruneDryOptions {
+	/**
+	 * Whether 'pruned' is returned, discouraged for large guilds.
+	 * @since 0.0.1
+	 * @default !this.large
+	 */
 	compute_prune_count: boolean;
 }
+
+/**
+ * The vanity URL retrieved from {@link Guild#fetchVanityURL}.
+ * @since 0.0.1
+ * @see https://discord.com/developers/docs/resources/guild#get-guild-vanity-url-example-partial-invite-object
+ */
+export interface GuildVanityURL {
+	/**
+	 * The code of this invite.
+	 * @since 0.0.1
+	 * @example "discord"
+	 */
+	code: string;
+
+	/**
+	 * The amount of uses this invite has.
+	 * @since 0.0.1
+	 * @example 42
+	 */
+	uses: number;
+}
```

Comment:
Should be moved to @klasa/dapi-types

Reply:
I think it's good to have it here, for documentation purposes.
If anything you can do `APIGuildVanityURLData` and I'll make this interface `implement` it.

Reply:
Actually, since typedocsgen will eventually point to `@klasa/rest`'s docs when you click on them, this should be good enough. Yeah, go ahead and port it to it.